### PR TITLE
feature/fbfw-139-incorporate-builds-for-st-petersburg

### PR DESCRIPTION
**Pull request by bryancodazen**
### Summary
This PR adds the ability for the user to build armies and fleet on the St. Petersburg territory. The STP territory is a map oddity because it is a home territory with two coasts. This means the Russia player can build fleets on the north/south coasts, and armies in the territory. 

### Video
https://drive.google.com/file/d/17IjLXr515Gx910jqks_5AFUDXqgLiWVq/view?usp=sharing

### Testing Plan
1. Start up your local dev environment and proceed to a game, then proceed to the build phase of the game. 
2. If you have more supply centers than territories you will now be able to build new units on your home territories. 
3. Depending on the type of territory, you may be able to build armies and/or fleets. 

### Rules

1. Clicking on a territory with a build will cancel the build for the territory
2. Clicking on a valid territory will popup the build UI to allow the user to select what unit to build
3. To cancel a build UI in progress click anywhere except one of the build unit icons. 

**Inland Territories:**
These territories only allow armies. 
Screen Shot 2022-04-25 at 5 06 01 PM

**Coastal Territories**
These territories allow the user to build armies or fleets. 
Screen Shot 2022-04-25 at 5 06 10 PM

**St. Petersburg**
This territory allows armies or fleets in their correct position. 
Screen Shot 2022-04-25 at 5 05 54 PM

### CODE REVIEW

File: beta-src/src/state/game/game-api-slice.ts [hunk 1]
```diff
@@ -164,43 +164,34 @@ const resetOrder = (state) => {
...
 ) => {
   const {
     data: { data: gameData },
   } = current(state);
   const { currentOrders } = gameData;
-  const orderForUnit = currentOrders.find((order) => {
-    return order.unitID === unitID;
-  });
+  let id = orderID;
+  if (!orderID) {
```
Comment (mariomelchor) [inline beta-src/src/state/game/game-api-slice.ts:176]: shouldn't this check be `if(!id)` instead since you'r setting the variable above on L175

Comment (bryancodazen) [reply beta-src/src/state/game/game-api-slice.ts:176]: Yea this code is kinda weird, I'll update it.

File: beta-src/src/interfaces/map/TerritoryMapData.ts [hunk 2]
```diff
@@ -1,16 +1,8 @@
-import {
-  AbsoluteCoordinates,
-  BBox,
-  Coordinates,
-  Label,
-  Territory,
-  Texture,
-  UnitSlot,
-} from "..";
+import { BBox, Coordinates, Label, Territory, Texture, UnitSlot } from "..";
```
Comment (mariomelchor) [inline beta-src/src/interfaces/map/TerritoryMapData.ts:1]: is this import correct?

**Review by mariomelchor — APPROVED** [review]
Just a comment otherwise LGTM

File: beta-src/src/interfaces/map/TerritoryMapData.ts (hunk 2 above)
Comment (bryancodazen) [reply beta-src/src/interfaces/map/TerritoryMapData.ts:1]: yea, just removed AbsoluteCoordinates as that interface was deleted from the codebase.

**Pull request merged**